build/templates/session.py.mako

```diff
@@ -190,6 +190,9 @@ constructor_params = helper.filter_parameters(init_function, helper.ParameterUsa
             raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, key))
         object.__setattr__(self, key, value)
 
+    def __getitem__(self, key):
```

Comment:
Is this needed? When you try to index a class that doesn't support it, the error is:
    
    TypeError: 'Foo' object does not support indexing

We wouldn't want this to be different. Let's let Python do its thing.

If you think the text in the exception is a good enough reason (I'm not convinced) then let's make it really useful:

* raise TypeError
* text should say `'Session' object does not support indexing. XXX
* XXX should be something that talks about the repeated capability containers for the class. The magic of codegen! (or we could at runtime figure it out)

Regardless, I'm not sure it's worth it.